vowpalwabbit/metrics.cc

```diff
@@ -72,6 +72,8 @@ void list_to_json_file(dsjson_metrics* ds_metrics, const std::string& filename,
       writer.String(ds_metrics->LastEventTime.c_str());
       writer.Key("dsjson_sum_cost_original");
       writer.Double(ds_metrics->DsjsonSumCostOriginal);
+      writer.Key("dsjson_sum_cost_original_first_slot");
```

Comment:
what exactly is the difference between dsjson_sum_cost_original and dsjson_sum_cost_original_first_slot? Do we ever need both at the same time?

Reply:
1) If we don't need both at the same time, an if else statement depending on cb vs ccb might be good. 2) Also I would add to the name dsjson_sum_cost_original_first_slot the keyword ccb.

Reply:
synced offline

Reply:
Addressed your comments. 
Keeping the metric name as it is (more generic) since we could possibly use this for slates. Slates is not a requirement right now by APS. Emitting the metric only when it is ccb required by APS.

Reply:
this is the stack for slates: gd, generate_interactions, scorer-identity, csoaa_ldf-rank, cb_adf,
cb_explore_adf_greedy, cb_sample, shared_feature_merger,
ccb_explore_adf, slates

Reply:
makes sense